presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -149,6 +155,16 @@ class ResponseHandler : public proxygen::HTTPTransactionHandler {
 
   void onBody(std::unique_ptr<folly::IOBuf> chain) noexcept override {
```

Comment:
I feel hard code type conditions inside http client is not a good idea. Can we find a way to do it dynamically?